File: src/main/java/github/macro/pieces/Lancer.java
package github.macro.pieces;

import github.macro.Direction;
import github.macro.players.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Macro303 on 2021-Jun-02.
 */
public class Lancer extends Piece {
	public Lancer(@NotNull Player player) {
		super(player, 10, "L", List.of(
				Direction.NORTH_EAST,
				Direction.SOUTH_EAST,
				Direction.SOUTH_WEST,
				Direction.NORTH_WEST
		));
	}

	@Override
	public String toString() {
		return "Lancer{} " + super.toString();
	}
}

File: src/main/kotlin/github/macro/pieces/Lancer.kt
package github.macro.pieces

import github.macro.Direction
import github.macro.players.Player

/**
 * Created by Macro303 on 2018-02-09.
 */
class Lancer(player: Player) : Piece(
	player = player,
	distance = 10,
	symbol = "L",
	directions = arrayOf(Direction.NORTH_EAST, Direction.SOUTH_EAST, Direction.SOUTH_WEST, Direction.NORTH_WEST)
) {
//	private static final String symbol = "\uD83C\uDFC7";
//	private static final String symbol = "🏇";
//	private static final String symbol = "L";

	override fun toString(): String = "Lancer() ${super.toString()}"
}